testapi.pm

```diff
@@ -1191,25 +1191,33 @@ sub get_test_data {
 
 =head2 validate_script_output
 
-  validate_script_output($script, $code | $regexp [, timeout => $timeout] [,quiet => $quiet])
+  validate_script_output($script, $code | $regexp [, timeout => $timeout] [, quiet => $quiet] [, title => $title] [, fail_message => $fail_message])
 
 Deprecated mode
 
   validate_script_output($script, $code, [$wait])
 
-Wrapper around script_output, that runs a callback on the output, or
-alternatively matches a regular expression. Use it as
+Similar to script_output, but this runs a callback on the output, 
+or alternatively matches a regular expression. Use it as
 
   validate_script_output "cat /etc/hosts", sub { m/127.*localhost/ };
   validate_script_output "cat /etc/hosts", qr/127.*localhost/;
+  validate_script_output "cat /etc/hosts", qr/127.*localhost/, title => 'localhost check', fail_message => 'No localhost line in /etc/hosts!';
   validate_script_output "cat /etc/hosts", sub { $_ !~ m/987.*somehost/ };
 
 =cut
 
 sub validate_script_output {    # no:style:signatures
     my ($script, $check, @args) = @_;
+    my %args = compat_args(
+        {
+            title => 'validate_script_output',
+            timeout => undef,
+            fail_message => 'output not validating',
+            quiet => 1,    # less noise by default
+        }, ((@args % 2) == 0) ? [] : ['timeout'], @args);
```

Comment:
I think this `compat_args()` call from L1212-L1218 can be moved below the call to `script_output()` and it should in theory preserve the old functionality. After all, the new named parameters are not needed by neither `testapi::script_output()` nor by `distribution::script_output()`

Reply:
as I tried to explain here: https://github.com/os-autoinst/os-autoinst/pull/2168/files#r968215741 it would make problems if someone call this function like:
```perl
validate_script_output($script, $check, title => "BLA", timeout => 3);
```

Reply:
As you specified now all parameters. I do not see any need for this workaround...
```suggestion
        }, ['timeout'], @args);
```